runtime/src/identity.rs

```diff
@@ -196,9 +202,11 @@ decl_module! {
             // This should never happen but just being defensive here
             ensure!(!<DidRecords>::exists(did), "DID must be unique");
             // 1.4. Signing keys can be linked to the new identity.
-            for sig_key in &signing_keys {
-                if !Self::can_key_be_linked_to_did( &sig_key.key, sig_key.key_type){
-                    return Err("One signing key can only belong to one DID");
+            for s_item in &signing_items {
+                if let Signer::Key(ref key) = s_item.signer {
```

Comment:
Can we use the eq function of signing_item here? That will also check the signing item type.

Reply:
This case is not a comparison.
The idea is that every signing item based on a key _HAS TO_ be compliant with 1-to-1 key to DID relation which is checked by `fn can_key_be_linked_to_did`. And it is double-checked if and only if `signer` is a key.

Reply:
My bad, I read that wrong.